File: Tema 2/Ejercicios ecuacion de segundo grado distintos lenguajes/Segundogrado.py
import time
import math

n = 200
a = [1.0] * n
b = [-5.0] * n
c = [6.0] * n
resultados = [(0.0, 0.0)] * n

inicio = time.perf_counter()

for i in range(n):
    disc = b[i] * b[i] - 4.0 * a[i] * c[i]
    x1 = (-b[i] + math.sqrt(disc)) / (2.0 * a[i])
    x2 = (-b[i] - math.sqrt(disc)) / (2.0 * a[i])
    resultados[i] = (x1, x2)

fin = time.perf_counter()
tiempo_ms = (fin - inicio) * 1000

print(f"Python - Tiempo de cálculo: {tiempo_ms:.4f} ms")
print(f"Control (Último resultado): x1 = {resultados[-1][0]}, x2 = {resultados[-1][1]}")

File: Tema 2/Ejercicios ecuacion de segundo grado distintos lenguajes/Segundogrado.js
const n = 200;
const a = new Float64Array(n).fill(1.0);
const b = new Float64Array(n).fill(-5.0);
const c = new Float64Array(n).fill(6.0);
const resultados_x1 = new Float64Array(n);
const resultados_x2 = new Float64Array(n);

const inicio = performance.now();

for (let i = 0; i < n; i++) {
    const disc = b[i] * b[i] - 4.0 * a[i] * c[i];
    resultados_x1[i] = (-b[i] + Math.sqrt(disc)) / (2.0 * a[i]);
    resultados_x2[i] = (-b[i] - Math.sqrt(disc)) / (2.0 * a[i]);
}

const fin = performance.now();
console.log(`JavaScript - Tiempo de cálculo: ${(fin - inicio).toFixed(4)} ms`);
console.log(`Control (Último resultado): x1 = ${resultados_x1[n-1]}, x2 = ${resultados_x2[n-1]}`);